### ISSUE TITLE
Append jump

### ISSUE THREAD

**Issue by CaveNightingale** [SpongePowered/Mixin#397]
I want to modify something like this  
```
public void doSomething(){
doA();  
doB();  
doC();  
doD();  
doE();  
}
```
```
public void doSomething(){
doA();  
if(handle_something()){  
doB();  
doC();  
}  
doD();  
doE();  
}
```

**Comment by Barteks2x** [comment]
While it's not exactly the same result, you can already do that by redirecting doB() and doC(), something like this (syntax and annotation details definitely wrong as writing without IDE or checking documentation):
```java
@Redirect(method = "doSomething", at = @At(value = "METHOD", target = "Lsome/package/ClassName;doB()V"))
private void handleDoB() {
    if (handle_something()) {
        doB();
        doC();
    }
}
// no-op
@Redirect(method = "doSomething", at = @At(value = "METHOD", target = "Lsome/package/ClassName;doC()V"))
private void handleDoC() {
}
```

**Comment by Mumfrey** [comment]
Unfortunately this kind of functionality comes with [similar usability considerations to rerouters](https://github.com/SpongePowered/Mixin/issues/365#issuecomment-539464542) but also the added consideration that the rules for where this kind of thing could happen get arbitrarily complex.

Effectively in the simple case this is very trivial, but even in the next-most-complex case are suddenly very complicated, and it just gets worse after that.

Consider:

```java
void doNothing(int a) {}

void myTarget()
{
   doNothing(1);
   // we want to jump here
   doNothing(2);
   doNothing(3);
   // to here
   doNothing(4);
}
```

Since `doNothing` is stack neutral, you can figure out the injection points to inject before `doNothing(2)` and it's obviously **after** `doNothing(1)` (we can't inject "before" `doNothing(2)` because the `this` and the `2` would remain on the stack).

* `@At(value = "INVOKE", target="doNothing(I)V", ordinal = 0, shift=Shift.AFTER)` // from
* `@At(value = "INVOKE", target="doNothing(I)V", ordinal = 3)` // to

Looks correct right? Well it's not.

```java
ALOAD 0
ILOAD 1
INVOKE doNothing
// from here, after first doNothing
ALOAD 0
ILOAD 2
// not here, the params for doNothing are on the stack
INVOKE doNothing
ALOAD 0
ILOAD 3
INVOKE doNothing
ALOAD 0
ILOAD 4
// to here, oh shit, we actually meant after 3, not before 4
INVOKE doNothing
```

* `@At(value = "INVOKE", target="doNothing(I)V", ordinal = 0, shift=Shift.AFTER)` // from
* `@At(value = "INVOKE", target="doNothing(I)V", ordinal = 2, shift=Shift.AFTER)` // to

```java
ALOAD 0
ILOAD 1
INVOKE doNothing
// from here, after first doNothing
ALOAD 0
ILOAD 2
// not here, the params for doNothing are on the stack
INVOKE doNothing
ALOAD 0
ILOAD 3
INVOKE doNothing
// to HERE, whew, that was not really obvious was it
ALOAD 0
ILOAD 4
INVOKE doNothing
```

What do the frames look like at our new (fixed) from and to:

```java
From: FRAME SAME
To: FRAME SAME
```

Whew. **Isn't that convenient, sure would be sodding impossible if the frames were contaminated. Fortunately they aren't so the VM won't explode**.

Wait...
what are you doing...
oh no...

```java
void myTarget()
{
   doNothing(1);
   // we want to jump here
   doNothing(2);
   int i = 3;
   doNothing(i++);
   // to here but we can't, and mixin has no way to fix this now. It can just tell you "no".
   doNothing(i);
}
```

Well, damn. Mixin can detect that this is utterly stupid, but it can't tell you how to fix it. In this case that's really obvious why not, but what about this:

```java
void myTarget()
{
   doNothing(1);
   // we want to jump here
   doNothing(getSomeValue(this.someInternalState));
   int i = 3;
   doNothing(i++);
   if (this.someInternalState) {
      i+=10;
   }
   // this is still here, but we now have a stack map frame here
   doNothing(getSomeValue(someInternalState);
   doSomething(i);
}
```

So basically, no. It's way too brittle for anything useful at a *high* level, and if you already KNOW the low level then you can just use a transformer with like 10 lines of code. Basically there's no **benefit** to having mixin provide this functionality because you have to know so much about the bytecode, and express the transformation in such a ridiculously complicated way, and mixin has to validate **so much state** that it just becomes a complete waste of time.

I've said it multiple times: Mixin is not "the one library" and I generally want it to solve problems that it can solve well. This is not something mixin can solve well, and so the official answer for this kind of request is always going to be "this makes more sense as a transformer".

**Comment by Mumfrey** [comment]
With the obvious caveat of "that doesn't mean there aren't other ways to do what you want *with* mixin, just that this functionality *as requested* won't be added.
